### PR TITLE
PUBDEV-5442-check makeGLMModel coefficients

### PR DESCRIPTION
**Pull request by wendycwong**
This PR fixes the bug in JIRA: https://0xdata.atlassian.net/browse/PUBDEV-5442?filter=-1

I have added a check to make sure the coefficients passed to makeGLMModel equals to the coefficients length of the GLM model.

### CODE REVIEW

File: h2o-algos/src/main/java/hex/api/MakeGLMModelHandler.java [hunk 1]
```diff
@@ -28,6 +28,16 @@ public GLMModelV3 make_model(int version, MakeGLMModelV3 args){
       throw new IllegalArgumentException("missing source model " + args.model);
     String [] names = model._output.coefficientNames();
     Map<String,Double> coefs = model.coefficients();
+    if (args.beta.length != names.length) {
+      StringBuffer sbuff = new StringBuffer();
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/api/MakeGLMModelHandler.java:32]: StringBuilder is better than StringBuffer

same can also be accomplished with String.join, eg. : "model coefficients needed are: \n" + String.join('\n', names)

Comment (wendycwong) [reply h2o-algos/src/main/java/hex/api/MakeGLMModelHandler.java]: Thanks Michal, I used String.join("/n", names).

File: h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    f2 = m2.predict(d)  # predict with model out of makeGLMModel
+    pyunit_utils.compare_frames_local(f1[1],f2[1],prob=1)
+    coefs = r['coefficients'][0]
+    coefs['wendy_dreams']=8
+    
+    try:
+        mw = glm.makeGLMModel(model=m,coefs=coefs)
+        sys.exit("Should have throw exception of bad coefficient length")
+    except Exception as ex:
+        print(ex)
```
Comment (honzasterba) [inline h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442.py:25]: should assert on execption message at least

File: h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    m2 = glm.makeGLMModel(model=m,coefs=r['coefficients'][0])
+    f1 = m.predict(d)   # predict with original model
+    f2 = m2.predict(d)  # predict with model out of makeGLMModel
+    pyunit_utils.compare_frames_local(f1[1],f2[1],prob=1)
+    coefs = r['coefficients'][0]
+    coefs['wendy_dreams']=8
+    
+    try:
+        mw = glm.makeGLMModel(model=m,coefs=coefs)
+        sys.exit("Should have throw exception of bad coefficient length")
```
Comment (honzasterba) [inline h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442.py:23]: use assert False, "message"

File: h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442_gaussian.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    m.train(training_frame=d,x=myX,y= myY)
+    r = glm.getGLMRegularizationPath(m)
+    m2 = glm.makeGLMModel(model=m,coefs=r['coefficients'][0])
+    f1 = m.predict(d)   # predict with original model
+    f2 = m2.predict(d)  # predict with model out of makeGLMModel
+    pyunit_utils.compare_frames_local(f1, f2 ,prob=1)
+    coefs = r['coefficients'][0]
+    coefs['wendy_dreams']=8
+    
+    try:
```
Comment (honzasterba) [inline h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442_gaussian.py:23]: same as above

File: h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442.py (hunk 2 above)
Comment (wendycwong) [reply h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442.py:25]: Done.  Thanks!

File: h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442_gaussian.py (hunk 4 above)
Comment (wendycwong) [reply h2o-py/tests/testdir_algos/glm/pyunit_glm_makeGLMModel_PUBDEV_5442_gaussian.py:23]: Done.

**Pull request merged**